PSReadLine/Accessibility.cs

```diff
@@ -10,14 +11,73 @@ internal class Accessibility
     {
         internal static bool IsScreenReaderActive()
         {
-            bool returnValue = false;
-
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                return IsAnyWindowsScreenReaderEnabled();           // Check for macOS VoiceOver
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                return IsVoiceOverEnabled();
+
+            // TODO: Support Linux per https://code.visualstudio.com/docs/configure/accessibility/accessibility
+            return false;
+        }
+
+        private static bool IsAnyWindowsScreenReaderEnabled() {
+            // The official way to check for a screen reader on Windows is
+            // SystemParametersInfo(SPI_GETSCREENREADER, ...) but it doesn't
+            // detect the in-box Windows Narrator and is otherwise known to be
+            // problematic.
+            //
+            // Unfortunately, the alternative method used by Electron and
+            // Chromium, where the relevant screen reader libraries (modules)
+            // are checked for does not work in the context of PowerShell
+            // because it relies on those applications injecting themselves into
+            // the app. Which they do not because it's not a windowed app, so
+            // we're stuck using the known-to-be-buggy way.
+            bool spiScreenReader = false;
+            PlatformWindows.SystemParametersInfo(PlatformWindows.SPI_GETSCREENREADER, 0, ref spiScreenReader, 0);
+            if (spiScreenReader)
+                return true;
+
+            // At least we can correctly check for Windows Narrator using the
+            // NarratorRunning mutex (which is mostly not broken with
+            // PSReadLine, as it were).
+            if (PlatformWindows.IsMutexPresent("NarratorRunning"))
+                return true;
+
+            return false;
+        }
+
+        private static bool IsVoiceOverEnabled()
+        {
+            try
+            {
+                // Use the 'defaults' command to check if VoiceOver is enabled
+                // This checks the com.apple.universalaccess preference for voiceOverOnOffKey
+                ProcessStartInfo startInfo = new()
+                {
+                    FileName = "defaults",
+                    Arguments = "read com.apple.universalaccess voiceOverOnOffKey",
+                    UseShellExecute = false,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    CreateNoWindow = true
+                };
+
+                using Process process = Process.Start(startInfo);
+                process.WaitForExit(250);
+                if (process.ExitCode == 0)

```

Comment:
Nitpick, to avoid an `InvalidOperationException` in case the process doesn't finish within 250ms.
```suggestion
                if (process.HasExited && process.ExitCode == 0)

```